app/components/GuideReviewForm/index.js

```diff
@@ -0,0 +1,116 @@
+    }
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    // Update if...
+    const doUpdate = ((nextState.isGuided !== this.state.isGuided) //...switching between yes/no options (affects disabled status for Select component)
+      || (nextProps.guideOptions.length !== this.props.guideOptions.length) //...or guide list is updated (affects Select component's options)
+      || (nextProps.selectedGuideId !== this.props.selectedGuideId) //...or a guide is selected (affects Select component and disabled status for Stars component)
+    );
+
+    return doUpdate;
+  }
+
+  componentWillUpdate(nextProps) {
+    // Do another API call if, somehow, the options array's length changes
+    if(this.props.guideOptions.length && nextProps.guideOptions.length !== this.props.guideOptions.length) {
+      console.log('componentWillUpdate newAPIcall:', nextProps.guideOptions);
+      this.props.getGuideList();  // lifting up state: API call is handled by WriteReviewPage container
+    }
+  }
+
+  handleButtonClick(e) {
+    // There are edge cases where clicking on the edge of a button can give undefined value
+    const isGuided = e.target.firstChild.value ? e.target.firstChild.value === "guided" : this.state.isGuided;
+
+    this.setState({
+      isGuided: isGuided,
+    });
+
+    // If "No" is selected, clear the Select component (this disables the rating as well)
+    if(!isGuided) {
+      this.props.handleSelectGuide(null);
+    }
+    else if(isGuided && !this.props.guideOptions.length) {
+      // Do API call if "yes" is selected and options array is currently empty
+      this.props.getGuideList();
+    }
+  }
+
+  render() {
+    return (
+      <PageSection title="Was the trip guided?">
+        {/* Yes/No buttons */}
+        <div className="btn-group-lg btn-group-toggle d-flex mx-3" data-toggle="buttons" onClick={this.handleButtonClick}>
+          <label className="btn btn-block mr-3 btn-success">
+            <input type="radio" value="guided" autoComplete="off"/>
+            Yes
+          </label>
+          <label className="btn btn-block mt-0 btn-danger">
+            <input type="radio" value="notGuided" autoComplete="off"/>
+            No
+          </label>
+        </div>
+        {/* Guide-list dropdown menu */}
+        <Select className="my-2"
+          disabled={!this.state.isGuided}
+          required={this.state.isGuided}
+          value={this.props.selectedGuideId}
+          onChange={this.props.handleSelectGuide}
+          options={this.props.guideOptions}
+          clearable={true}
+        />
+        {/* Rate using Stars component */}
+        <div className="d-flex flex-row justify-content-around">
+          <label className={"d-flex align-self-center mb-0 " + (!!this.props.selectedGuideId ? "" : "text-secondary")}>
+            Rate your guide:
+          </label>
+          <Stars className="d-flex"
+            size={35}
+            editable={!!this.props.selectedGuideId}
```

Comment:
This can be `isGuided` right? Not a fan of `!!`

Reply:
Editable needs to be a bool, and `!!` is a trick for conversion that I'm actually a fan of. :-P This specifically enables a rating to be made after a guide is selected rather than just "Yes."

Reply:
`I'm a fan of `Boolean(this.props.selectedGuideId)`

Reply:
And I'm not a fan of ^that since typecasting is a big no-no in C++, which are my roots in programming, but I suppose this is fine for JS (probably does about the same thing as `!!` anyway).